packages/block-library/src/paragraph/edit.js

```diff
@@ -23,6 +23,11 @@ import {
 import { createBlock } from '@wordpress/blocks';
 import { formatLtr } from '@wordpress/icons';
 
+/**
+ * Internal dependencies
+ */
+import { useExitOnEnterAtEnd } from './use-enter';
```

Comment:
As the name suggests this only exits the block at the end, but I'm wondering if we also want to "split" it in the middle if we're in an empty paragraph but in the middle. (Like lists). Any thoughts @nosolosw

Reply:
It seems good with split in the middle and since you have pushed changes for this, we should rename the block supports and hook.

Also noting that it seems kind of weird to have a generic named block support but is implemented in `paragraph` which is specific. I'm wondering if we could do this in RichText directly, similar to the `onSplit` functions and utilize `default block`? 🤔